lib/rust/ensogl/component/slider/src/lib.rs

```diff
@@ -0,0 +1,455 @@
+/// `STEP_SIZE` pixels and adjust the slider value more quickly. Dragging the mouse downwards will
+/// increase the precision and change the value more slowly.
+const PRECISION_ADJUSTMENT_STEP_SIZE: f32 = 50.0;
+/// The actual slider precision changes exponentially with each adjustment step. When the adjustment
+/// is changed by one step, the slider's precision is changed to the next power of `STEP_BASE`. A
+/// `STEP_BASE` of 10.0 results in the precision being powers of 10 for consecutive steps, e.g [1.0,
+/// 10.0, 100.0, ...] when decreasing the precision and [0.1, 0.01, 0.001, ...] when increasing the
+/// precision.
+const PRECISION_ADJUSTMENT_STEP_BASE: f32 = 10.0;
+/// The maximum number of digits after the decimal point to be displayed when showing the
+/// component's value.
+const VALUE_DISPLAY_MAX_DIGITS_AFTER_DECIMAL: usize = 8;
+
+
+
+// ========================
+// === Slider component ===
+// ========================
+
+/// A slider UI component that allows adjusting a value through mouse interaction. Dragging the
+/// slider in a horizontal direction changes the value, limited to a range between `min_value` and
+/// `max_value`. The selected value is displayed, and a track fills the slider proportional to the
+/// value within the specified range. Dragging the slider in a vertical direction adjusts the
+/// precision of the slider. The precision affects the increments by which the value changes when
+/// the mouse is moved.
+#[derive(Debug, Deref)]
+pub struct Slider {
+    /// Public FRP api of the component.
+    #[deref]
+    pub frp: Frp,
+    model:   Rc<Model>,
+    /// Reference to the application the component belongs to. Generally required for implementing
+    /// `application::View` and initialising the `Model` and `Frp` and thus provided by the
+    /// component.
+    pub app: Application,
+}
+
+impl Slider {
+    /// Construct a new slider component.
+    pub fn new(app: &Application) -> Self {
+        let model = Rc::new(Model::new(app));
+        let app = app.clone_ref();
+        let frp = Frp::new();
+        Self { frp, model, app }.init()
+    }
+
+    fn init(self) -> Self {
+        let network = self.frp.network();
+        let input = &self.frp.input;
+        let output = &self.frp.private.output;
+        let model = &self.model;
+        let scene = &self.app.display.default_scene;
+        let mouse = &scene.mouse.frp;
+        let keyboard = &scene.keyboard.frp;
+        let track_pos_anim = Animation::new_non_init(network);
+        let background_color_anim = color::Animation::new(network);
+        let track_color_anim = color::Animation::new(network);
+        let value_text_color_anim = color::Animation::new(network);
+        let label_color_anim = color::Animation::new(network);
+
+        frp::extend! { network
+
+            // === User input ===
+
+            background_click <- model.background.events.mouse_down_primary.constant(());
```

Comment:
creating an FRP node, like with a `.constant(())` has a slight performance hit. If you just want to create a new shorter variable, use `let background_click = &model.background.events.mouse_down_primary;` instead.